site/server/app/com/fortysevendeg/exercises/controllers/ApplicationController.scala

```diff
@@ -35,8 +40,8 @@ class ApplicationController(userService: UserServices)(implicit exerciseOps: Exe
     exerciseOps.getLibraries.runTask match {
       case \/-(libraries) ⇒
         request.session.get("oauth-token") map { token ⇒
-          val user = userService.getUserByLogin(GetUserByLoginRequest(login = request.session.get("user").getOrElse("")))
-          user.map(response ⇒ Ok(views.html.templates.home.index(user = response.user, libraries = libraries)))
+          val user = userService.getUserByLogin(request.session.get("user").getOrElse(""))
```

Comment:
If `userService.getUserByLogin` returns an `Option` it means that it may be blocking. In which case the default executor pool may not be appropriate. This should all be in the context of `Free` and monadically computed along with `getLibraries` above so it can be lifted to the appropriate interpreter target monad.